app/services/open_library.rb

```diff
@@ -0,0 +1,33 @@
+class OpenLibrary
+  require 'faraday'
+  require 'json'
+
+  def initialize(isbn)
+    @isbn = isbn
+  end
+
+  def fetch_data
+    response = books
+    if response != 'not found'
+      response = JSON.parse(response)
+      response = response[@isbn.to_s].slice('title', 'subtitle', 'authors', 'number_of_pages')
+      response['isbn'] = @isbn
+      response['authors'] = take_authors(response['authors'])
+      response
+    else
+      response
+    end
+  end
+
+  def books
+    uri = "https://openlibrary.org/api/books?bibkeys=#{@isbn}&format=json&jscmd=data"
```

Comment:
Use credentials to save the base `https://openlibrary.org/api` and merge with `/books` inside method, and remember to use the query params as JSON, you should define inside the initializer

```ruby
@options = { query: { bibkeys: "ISBN:#{isbn}", format: 'json', jscmd: 'data' } }

Faraday.get("#{base_url}/books", @options,
```